ProviderFactory/OpenRouteServiceFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the BazingaGeocoderBundle package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Bazinga\GeocoderBundle\ProviderFactory;
+
+use Geocoder\Provider\OpenRouteService\OpenRouteService;
+use Geocoder\Provider\Provider;
+use Http\Client\HttpClient;
+use Http\Discovery\HttpClientDiscovery;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class OpenRouteServiceFactory extends AbstractFactory
+{
+    protected static $dependencies = [
+        ['requiredClass' => OpenRouteService::class, 'packageName' => 'geocoder-php/openrouteservice-provider'],
+    ];
+
+    /**
+     * @phpstan-param array{api_key: string, httplug_client: ?HttpClient} $config
+     */
+    protected function getProvider(array $config): Provider
+    {
+        $httplug = $config['httplug_client'] ?: HttpClientDiscovery::find();
+
+        return new OpenRouteService($httplug, $config['api_key']);
+    }
+
+    protected static function configureOptionResolver(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'httplug_client' => null,
+            'api_key' => null,
+        ]);
+
+        $resolver->setAllowedTypes('httplug_client', [HttpClient::class, 'null']);
+        $resolver->setAllowedTypes('api_key', ['string']);
```

Comment:
As `api_key` is required in the `OpenRouteService`, we should configure it as required :)

Reply:
Good catch, thanks! I required it in [f5c6db4](https://github.com/geocoder-php/BazingaGeocoderBundle/pull/329/commits/f5c6db4b4ae4b019887279f37821f674e34e235f)